src/test/java/seedu/address/logic/commands/SortStudentCommandTest.java

```diff
@@ -55,46 +65,148 @@ public void execute_validParameter_success() {
+
+        // setup new model for testing correctness of sort
+        Person firstEmail = new PersonBuilder().withEmail("a@example.co").withStudentId("A1234567Z").build();
+        Person secondEmail = new PersonBuilder().withEmail("a@example.com").withStudentId("A1234567F").build();
+        Person thirdEmail = new PersonBuilder().withEmail("aaa@example.com").withStudentId("A1234567A").build();
+        Person fourthEmail = new PersonBuilder().withEmail("b@anotheremail.com").withStudentId("A1234567G").build();
+        Person fifthEmail = new PersonBuilder().withEmail("b@example.com").withStudentId("A1234567L").build();
+
+        // add persons in non sorted order
+        model = new ModelManager();
+        model.addPerson(fifthEmail);
+        model.addPerson(secondEmail);
+        model.addPerson(fourthEmail);
+        model.addPerson(firstEmail);
+        model.addPerson(thirdEmail);
+
+        // add persons in sorted order
+        expectedModel = new ModelManager();
+        expectedModel.addPerson(firstEmail);
+        expectedModel.addPerson(secondEmail);
+        expectedModel.addPerson(thirdEmail);
+        expectedModel.addPerson(fourthEmail);
+        expectedModel.addPerson(fifthEmail);
+
+        try {
+            // sort by email
+            SortStudentCommand command = new SortStudentCommand(SortStudentParameter.EMAIL);
+            command.execute(model);
+            ObservableList<Person> expectedPersonList = expectedModel.getAddressBook().getPersonList();
+            ObservableList<Person> actualPersonList = model.getAddressBook().getSortedPersonList();
+            assertEquals(expectedPersonList, actualPersonList);
+        } catch (CommandException e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void execute_sortByStudentId_correctSorting() {
+
+        // setup new model for testing correctness of sort
+        Person firstId = new PersonBuilder().withStudentId("A1123456A").withEmail("a@example.com").build();
+        Person secondId = new PersonBuilder().withStudentId("A1234566B").withEmail("h@example.com").build();
+        Person thirdId = new PersonBuilder().withStudentId("A1234567A").withEmail("k@example.com").build();
+        Person fourthId = new PersonBuilder().withStudentId("A1234567B").withEmail("g@example.com").build();
+        Person fifthId = new PersonBuilder().withStudentId("A2976237K").withEmail("f@example.com").build();
+
+        // add persons in non sorted order
+        model = new ModelManager();
+        model.addPerson(firstId);
+        model.addPerson(secondId);
+        model.addPerson(thirdId);
+        model.addPerson(fourthId);
+        model.addPerson(fifthId);
+
+        // add persons in sorted order
+        expectedModel = new ModelManager();
+        expectedModel.addPerson(firstId);
+        expectedModel.addPerson(secondId);
+        expectedModel.addPerson(thirdId);
+        expectedModel.addPerson(fourthId);
```

Comment:
Do you think it would be better if you store the 5 different persons in an array first then iterate through the array to add them 1 by 1? I understand that this is an easier approach but can lead to extra lines of code, just something to consider.

Reply:
Yep. Will use an array instead, also makes code more clear.